overlord/state/task.go

```diff
@@ -393,6 +393,11 @@ func (t *Task) HaltTasks() []*Task {
 	return t.state.tasksIn(t.haltTasks)
 }
 
+// NumHaltTasks returns the number of tasks registered to wait for t.
+func (t *Task) NumHaltTasks() int {
```

Comment:
can we have add somewhere in the current tests for this pkg a simple test for this?